### PR TITLE
Allow custom minute interval setting for UIDatePicker with date and time, and time of day answer formats

### PR DESCRIPTION
**Pull request by MacroYau**

### CODE REVIEW

File: ResearchKit/Common/ORKAnswerFormat.h [hunk 1]
```diff
@@ -1191,7 +1191,8 @@ ORK_CLASS_AVAILABLE
  
  @return An initialized time of day answer format.
  */
-- (instancetype)initWithDefaultComponents:(nullable NSDateComponents *)defaultComponents NS_DESIGNATED_INITIALIZER;
+- (instancetype)initWithDefaultComponents:(nullable NSDateComponents *)defaultComponents
+                           minuteInterval:(NSInteger)minuteInterval NS_DESIGNATED_INITIALIZER;
```
[the remark this answers is not in the document]
Comment (MacroYau) [reply ResearchKit/Common/ORKAnswerFormat.h]: Oh thanks for the heads up! Removed those unnecessary modifications to the constructors.

File: ResearchKit/Common/ORKAnswerFormat.m [hunk 2]
```diff
@@ -1277,12 +1278,17 @@ - (NSDate *)pickerDefaultDate {
     return ORKTimeOfDayDateFromComponents(newDateComponents);
 }
 
+- (NSInteger)pickerMinuteInterval {
```
Comment (srinathtm-apple) [inline ResearchKit/Common/ORKAnswerFormat.m]: is this method necessary? I don't see it being referenced anywhere..

File: ResearchKit/Common/ORKAnswerFormat.m [hunk 3]
```diff
@@ -1418,6 +1428,10 @@ - (NSDate *)pickerMaximumDate {
     return self.maximumDate;
 }
 
+- (NSInteger)pickerMinuteInterval {
```
Comment (srinathtm-apple) [inline ResearchKit/Common/ORKAnswerFormat.m]: is this method necessary? I don't see it being referenced anywhere..

File: ResearchKit/Common/ORKAnswerFormat.m (hunk 2 above)
Comment (MacroYau) [reply ResearchKit/Common/ORKAnswerFormat.m]: Thanks for catching this! I added them at the first place, but seems they are extraneous now. Having the `minuteInterval` instance variable alone should do. Removed the getters.

**Comment by srinathtm-apple** [comment]
Merging into master.

Thanks for the submission @MacroYau

**Pull request merged**